Add note about `throwHttpErrors` and retry functionality (#368)
sindresorhus/ky: readme.md

Setting this to `false` may be useful if you are checking for resource availability and are expecting error responses.

> Note: If `false`, error responses are considered successful, therefore no retry will have place.

Reviewer: ```suggestion
Note: If `false`, error responses are considered successful and the request will not be retried.
```

> Note: Setting this to `false` will skip the retry functionality for non-2xx status codes because the response is considered successful, therefore no need to retry.

Reviewer 2: ```suggestion
Note: If `false`, error responses are considered successful, therefore no retry will have place.
```
